quarkus-micro-image.yaml

```diff
@@ -0,0 +1,25 @@
+- name: ubi
+  version: latest
+  from: registry.access.redhat.com/ubi8/ubi-minimal
+
+# need to rename from image into "scratch" to disable /tmp cleanup
+- name: scratch
+  version: latest
+  from: registry.access.redhat.com/ubi8/ubi-micro
+
+- name: quay.io/quarkus/quarkus-distroless-image
+  version: latest
+  from: scratch
+  artifacts:
+    - name: libgcc_s.so.1
+      image: ubi
+      path: /usr/lib64/libgcc_s.so.1
+      dest: /usr/lib64
+    - name: libstdc++.so.6
```

Comment:
I didn't know we need libstdc++ (except when using UPX). Do you know if this is a new requirement?

Reply:
Here is what my application needs:
```
root@aed5a43099fa:/# ldd application
        linux-vdso.so.1 (0x00007ffd29b1c000)
        libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007fc4c3baa000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fc4c3a5b000)
        libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007fc4c3a3f000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fc4c3a1c000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fc4c3a16000)
        librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007fc4c3a0b000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fc4c3817000)
        /lib64/ld-linux-x86-64.so.2 (0x00007fc4c3d90000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fc4c37fc000)
```

Reply:
One solution could be to build in native and then we could use the smallest possible distroless (static).
https://www.graalvm.org/reference-manual/native-image/StaticImages/

Reply:
Thanks for the list. I believe it makes sense to include it. As I'm integrating UPX, having a runtime image that provides it is great.

Reply:
This is why we use `gcr.io/distroless/cc` in https://github.com/quarkusio/quarkus-images/blob/main/quarkus-distroless-image.yaml because it includes `libgcc1 and its dependencies`:
https://github.com/GoogleContainerTools/distroless/blob/main/cc/README.md

Reply:
> I didn't know we need libstdc++ (except when using UPX). Do you know if this is a new requirement?

FWIW, any apps using the `sunec.a` (i.e. EC crypto) static library from OpenJDK will add a dep on `-lstdc++`.

Reply:
`-H:+TraceNativeToolUsage` is useful to inspect this.